File: leetcode/reverse-linked-list/reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* prev_node = NULL;
        while(head) {
            ListNode* next_node = head->next;
            head->next = prev_node;
            prev_node = head;
            head = next_node;
        }
        return prev_node;
    }
};

File: hackerrank/basic_data_structures/reverse_linkedlist.js
function reverse(head) {
    if (head == null) return null;

    let prev = null;
    while(head.next != null) {
        const nextNode = head.next;
        head.next = prev;
        prev = head;
        head = nextNode;
    }
    head.next = prev;
    return head;
}
